test/validation/api/ipsec/reass_test_vectors.h

```diff
@@ -0,0 +1,658 @@
+ /* Copyright (c) 2021, Marvell
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+#ifndef _ODP_REASS_TEST_VECTORS_H_
+#define _ODP_REASS_TEST_VECTORS_H_
+
+/*
+ * The header file includes below test vectors
+ *
+ *	1) pkt_ipv6_udp_p1
+ *		pkt_ipv6_udp_p1_f1
+ *		pkt_ipv6_udp_p1_f2
+ *
+ *	2) pkt_ipv6_udp_p2
+ *		pkt_ipv6_udp_p2_f1
+ *		pkt_ipv6_udp_p2_f2
+ *		pkt_ipv6_udp_p2_f3
+ *		pkt_ipv6_udp_p2_f4
+ *
+ *	3) pkt_ipv6_udp_p3
+ *		pkt_ipv6_udp_p3_f1
+ *		pkt_ipv6_udp_p3_f2
+ *		pkt_ipv6_udp_p3_f3
+ *		pkt_ipv6_udp_p3_f4
+ *		pkt_ipv6_udp_p3_f5
+ *
+ *	4) pkt_ipv4_udp_p4
+ *		pkt_ipv4_udp_p4_f1
+ *		pkt_ipv4_udp_p4_f2
+ *
+ *	5) pkt_ipv4_udp_p5
+ *		pkt_ipv4_udp_p5_f1
+ *		pkt_ipv4_udp_p5_f2
+ *		pkt_ipv4_udp_p5_f3
+ *		pkt_ipv4_udp_p5_f4
+ *
+ *	6)pkt_ipv4_udp_p6
+ *		pkt_ipv4_udp_p6_f1
+ *		pkt_ipv4_udp_p6_f2
+ *		pkt_ipv4_udp_p6_f3
+ *		pkt_ipv4_udp_p6_f4
+ *		pkt_ipv4_udp_p6_f5
+ */
+
+static const ODP_UNUSED ipsec_test_packet pkt_ipv6_udp_p1 = {
+	.len = 1514,
+	.l2_offset = 0,
+	.l3_offset = 14,
+	.l4_offset = 54,
+	.data = {
+		/* ETH */
+		0xf1, 0xf1, 0xf1, 0xf1, 0xf1, 0xf1,
+		0xf2, 0xf2, 0xf2, 0xf2, 0xf2, 0xf2, 0x86, 0xdd,
+
+		/* IP */
+		0x60, 0x00, 0x00, 0x00, 0x05, 0xb4, 0x11, 0x40,
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x00, 0x00, 0xff, 0xff, 0x0d, 0x00, 0x00, 0x02,
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x00, 0x00, 0xff, 0xff, 0x02, 0x00, 0x00, 0x02,
+
+		/* UDP */
+		0x08, 0x00, 0x27, 0x10, 0x05, 0xb4, 0x2b, 0xe8,
+		[62 ... 1513] = 0x58,
```

Comment:
This syntax seems to be a GCC extension which we want to avoid for portability.

Reply:
What else do you suggest? Populate the test vector while using it? It would mean we will have a differential treatment for reassembly vectors. If you are fine with that, I can make that change.

Reply:
One possibility is to just initialize all the needed array elements. With macros one could lower the number of code lines needed, but it gets a bit ugly:

e.g. #define B8(b) b, b, b, b, b, b, b, b
     #define B64(b) B8(b), B8(b), B8(b), B8(b), B8(b), B8(b), B8(b), B8(b)
     #define B256(b) B64(b), B64(b), B64(b), B64(b)

Instead of that I would probably make it so that the reassembly test vectors are preprocessed at test program start so that the trailing zeros in the packet data are replaced with the value of the last non-zero packet data byte. The function to do that could be in the same header file as the test vectors and be called right after startup, not within test case code. And there should be appropriate comments that make it clear that the static initialization is not all there is to it.

Or if that feels too hacky, one could use some sort of test_part factory for the reassembly test vectors so that the statically defined (and partially initialized) test parts are not even visible to the test case code. The factory could then do whatever initialization is needed without the need for complete initializers for everything.

Reply:
> Instead of that I would probably make it so that the reassembly test vectors are preprocessed at test program start so that the trailing zeros in the packet data are replaced with the value of the last non-zero packet data byte.

Will update so in v18.